ftd/src/ast/component.rs

```diff
@@ -441,7 +449,55 @@ impl Loop {
 
         let loop_statement = loop_header.value.string(doc_id)?;
 
-        let (on, alias) = ftd::ast::utils::split_at(loop_statement.as_str(), ftd::ast::utils::AS);
+        let is_for_loop = loop_header.key.eq(ftd::ast::utils::FOR);
+
+        let (alias, on, key_name) = if is_for_loop {
+            let (pair, on) =
+                ftd::ast::utils::split_at(loop_statement.as_str(), ftd::ast::utils::IN);
+
+            let on = if let Some(on) = on {
```

Comment:
Change the below unwrap to this 
```suggestion
            let on = on.ok_or(ftd::ast::Error::Parse {
                message: "Statement \"for\" needs a list to operate on".to_string(),
                doc_id: doc_id.to_string(),
                line_number: loop_header.line_number,
            })?;
```